src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -169,7 +169,17 @@ DistributedTableSize(Oid relationId, char *sizeQuery)
 							   " blocks which contain multi-shard data modifications")));
 	}
 
-	/* try to open relation, will error out if the relation does not exist */
+	/*
+	 * importantly, this also accepts cache invalidation messages, so we'll notice if the
+	 * relation has been removed concurrently
+	 */
+	LockRelationOid(relationId, AccessShareLock);
+
+	if (!SearchSysCacheExists1(RELOID, ObjectIdGetDatum(relationId)))
+	{
+		ereport(ERROR, (errmsg("can't find table size, relation does not exist")));
+	}
+
 	relation = relation_open(relationId, AccessShareLock);
```

Comment:
Maybe use `try_relation_open(relationId, AccessShareLock)`, see `pg_table_size`. For consistency, what if we also return NULL like `pg_table_size`?

Reply:
I'm also partial to this idea, but willing to punt on it to get the current PR merged so these tests stop failing every time e.g. a new function is added.

Reply:
For future archaeologists: I've [opened a ticket](https://github.com/citusdata/citus/issues/1705) for this